### ISSUE TITLE
OpenXR applications cannot use SteamVR

### ISSUE THREAD

**Issue by FamiliarBreakfast** [flathub/com.valvesoftware.Steam#1010]
**Game information**
I have only tested this with Godot's OpenXR and the OpenXR test program as I have no other programs with which to test. I believe this issue affects all OpenXR implementations.

**Distribution name and version where applicable**
Fedora Linux 37 
Kernel 6.0.7-301.fc37.x86_64

**Flatpak info**
Flatpak 1.14.0
default
host

**Problem description**
OpenXR applications cannot use SteamVR's OpenXR implementation. There are multiple parts to this problem:
1. steamxr_linux64.json is not in a standard location, so no existing OpenXR loader will detect it automatically.
2. If one forces OpenXR to load the configuration anyway, it will fail, presumably due to flatpak sandboxing.
This can be done through the `XR_RUNTIME_JSON` env variable.

Log file output from a simple test program: https://pastebin.com/0ZjCdy5e
Relevant excerpt:
```
Verbose [GENERAL | xrCreateInstance | OpenXR-Loader] : Entering loader trampoline
Verbose [GENERAL | xrCreateInstance | OpenXR-Loader] : Entering LoaderInstance::CreateInstance
Verbose [GENERAL | xrCreateInstance | OpenXR-Loader] : Entering loader terminator
Verbose [GENERAL | xrCreateInstance | OpenXR-Loader] : Completed loader terminator
Error [GENERAL | xrCreateInstance | OpenXR-Loader] : LoaderInstance::CreateInstance chained CreateInstance call failed
Error [GENERAL | xrCreateInstance | OpenXR-Loader] : LoaderInstance::CreateInstance chained CreateInstance call failed
Info [GENERAL | xrCreateInstance | OpenXR-Loader] : RuntimeInterface::UnloadRuntime - Unloading RuntimeInterface
Info [GENERAL |  | OpenXR-Loader] : RuntimeInterface being destroyed.
Error [GENERAL | xrCreateInstance | OpenXR-Loader] : xrCreateInstance failed
Error [GENERAL | xrCreateInstance | OpenXR-Loader] : xrCreateInstance failed
ERROR: Failed to create XR instance.
at: create_instance (modules/openxr/openxr_api.cpp:286)
```

**Does this issue reproduce with native Steam**
No.

**Comment by Vixea** [comment]
could you link the test program?

**Comment by Adamcake** [comment]
Yeah, doesn't work at all. Beat Saber is the most obvious test case.

**Comment by DomiStyle** [comment]
This is still an issue it seems. Here's a quick test application made in Godot: 
[OpenXrTest.zip](https://github.com/flathub/com.valvesoftware.Steam/files/13527365/OpenXrTest.zip)

Run `OpenXrTest.sh` or open `project.godot` in the Godot editor.

This will work when added as a non-steam game, it will fail when launched directly from the editor or via the .sh file.

![image](https://github.com/flathub/com.valvesoftware.Steam/assets/975131/3000f37f-f575-4ced-b3c5-2d723888b583)

**Comment by Chocolateandmilkwin** [comment]
Anybody have a workaround for this?
